controllers/search_controller.go

```diff
@@ -230,6 +223,25 @@ func (r *SearchReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 	once.Do(func() {
 		addon.CreateAddonOnce(ctx, instance)
 	})
+	cleanUpFunc := func() {
+		// delete legacy servicemonitor setup
+		result, err = r.deleteLegacyServiceMonitorSetup(instance)
+		cleanupComplete = true
+		log.Info("cleanupComplete. Setting cleanupComplete to ", "cleanupComplete", cleanupComplete)
+		if result != nil {
+			cleanupComplete = false
+			log.Info("cleanup not complete. Setting cleanupComplete to ", "cleanupComplete", cleanupComplete)
+			log.Error(err, "Failed to remove legacy ServiceMonitor setup ")
+		}
```

Comment:
Move the success log to the `else{}` to avoid printing both messages, which can be confusing.

```suggestion
		if result != nil {
			cleanupComplete = false
			log.Info("cleanup not complete. Setting cleanupComplete to ", "cleanupComplete", cleanupComplete)
			log.Error(err, "Failed to remove legacy ServiceMonitor setup ")
		} else {
			log.Info("cleanupComplete. Setting cleanupComplete to ", "cleanupComplete", cleanupComplete)
		}
```